src/app/ReadHandler.h

```diff
@@ -536,6 +545,12 @@ class ReadHandler : public Messaging::ExchangeDelegate
     uint16_t mMinIntervalFloorSeconds = 0;
     uint16_t mMaxInterval             = 0;
 
+    // When we close the read handler after it fails to resume subscription beacuse the session is not established,
+    // we should use these values to delete the subscription storage.
+    FabricIndex mResumingSubscriptionFabricIndex = kUndefinedFabricIndex;
+    NodeId mResumingSubscriptionNodeId           = kUndefinedNodeId;
+    bool mIsResumingSubscription                 = false;
```

Comment:
This is adding 24 bytes of space per ReadHandler, as far as I can tell, to store 10 bytes of information.

1. We can do better.  For example, I don't think we need an mIsResumingSubscription boolean at all, as long as we set the fabric index and node id fields properly.   But more importantly, we have the data in `HandleDeviceConnectionFailure`, no?  We should be passing it to the right code, instead of having to store it in members.
2. Shouldn't some of this be conditioned on CHIP_CONFIG_PERSIST_SUBSCRIPTIONS?